ksql-engine/src/main/java/io/confluent/ksql/security/KsqlAccessValidatorProvider.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.security;
+
+import io.confluent.ksql.exception.KsqlTopicAuthorizationException;
+import java.util.Collections;
+import java.util.Set;
+import org.apache.kafka.common.acl.AclOperation;
+
+/**
+ * An implementation of {@link KsqlAccessValidator} that provides authorization checks
+ * from the Kafka service.
+ */
+public class KsqlAccessValidatorProvider implements KsqlAccessValidator {
+  @Override
+  public void checkAccess(
+      KsqlSecurityContext securityContext,
+      String topicName,
+      AclOperation operation
+  ) {
+    final Set<AclOperation> authorizedOperations = securityContext.getServiceContext()
```

Comment:
should we pass in the serviceContext itself, if thats all is needed to perform the check?

Reply:
This is the impl of the interface which uses the `KsqlSecurityContext`. I cannot pass only the `ServiceContext.